Civi/Api4/Contact.php

```diff
@@ -96,8 +96,37 @@ public static function getDuplicates($checkPermissions = TRUE) {
    * @return Action\Contact\MergeDuplicates
    */
   public static function mergeDuplicates($checkPermissions = TRUE) {
-    return (new Action\Contact\MergeDuplicates(__CLASS__, __FUNCTION__))
+    return (new Action\Contact\MergeDuplicates(self::getEntityName(), __FUNCTION__))
       ->setCheckPermissions($checkPermissions);
   }
 
+  protected static function getDaoName(): string {
+    // Child classes (Individual, Organization, Household) need this.
+    return 'CRM_Contact_DAO_Contact';
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public static function getInfo(): array {
+    $info = parent::getInfo();
+    $contactType = static::getEntityName();
+    // Adjust info for child classes (Individual, Organization, Household)
+    if ($contactType !== 'Contact') {
+      $info['icon'] = \CRM_Contact_BAO_ContactType::getContactType($contactType)['icon'] ?? $info['icon'];
+      $info['type'] = ['DAOEntity', 'ContactType'];
+      // This forces the value into get and create api actions
+      $info['where'] = ['contact_type' => $contactType];
```

Comment:
@totten this is the new mechanism that drives this whole thing. You can now define, at the api4-entity level, a `where` property which api4 will force into get, create and getFields. 

We could re-use this to, for example, create an `EventTemplate` api which extends `Event` but adds `$info['where'] = ['is_template' => TRUE];` and then that api would only ever get or create event templates.

Reply:
That's, pretty cool. It does help to give a general sense to the bigger PR. :+1: